### DISCUSSION TITLE
AOSP 12.0 v402

### DISCUSSION THREAD

**Discussion by phhusson** [phhusson/treble_experimentations#2174] [General]
- More February security patch level (previous one had only some patches, not all)
- Well this will allow users to test OTA on their devices

<hr /><em>This discussion was created from the release <a href='https://github.com/phhusson/treble_experimentations/releases/tag/v402'>AOSP 12.0 v402</a>.</em>

**Comment by dcn1961** [comment]
When I reboot to recovery it gets stuck at twrp boot screen and goes no further. It does this on PE as well but not Octavi. Only way to fix is to reinstall whole stock system and recovery and restore nandroid backup. Using Umidigi Power. Don't have this problem with any a11. Thanks.

**Comment by mrazzari** [comment]
Tried it after having just installed v400 successfully (and happily) on Moto G8 Plus. 
But v402 gets stuck in the boot animation. Tried `fastboot -w`. Don't know how to further debug. Returned to v400.

**Reply by phhusson** [reply]
there is no "v400" though

**Reply by mrazzari** [reply]
Sorry, meant the latest one, AOSP 12.0 v400.h. This boots on my Moto G8 plus. And 402 doesn't.

**Reply by Dunedan** [reply]
Same here with Moto G7 Play.

**Comment by SteQuOFFICIAL** [comment]
I understand that the problems on Pixel XL are still the same?

**Reply by phhusson** [reply]
yes

**Reply by SteQuOFFICIAL** [reply]
Got it, thank you for your answer.

**Comment by wilsenbaltazar** [comment]
You're awesome Man. Thanks for all your work!

**Comment by RendeRR** [comment]
OTA does not work. Displays that there is an update, downloads, asks to reboot. But after the reboot, the system is the same and again the message that there is an update.

**Reply by phhusson** [reply]
Device? Screenshot?

**Reply by RendeRR** [reply]
Hello.
Alldocube iplay 30
![Screenshot_20220209-192652](https://user-images.githubusercontent.com/790707/153246603-262de2a6-e9bb-4510-ae85-598fe86ff59b.png)
![Screenshot_20220209-192716](https://user-images.githubusercontent.com/790707/153246609-95126920-3ff1-4bf8-a937-1fcc6224e6e9.png)
![Screenshot_20220209-192724](https://user-images.githubusercontent.com/790707/153246612-c6f6bf27-3faf-477e-8f33-3646ac82b4e7.png)
![Screenshot_20220209-192931](https://user-images.githubusercontent.com/790707/153246616-87947771-4507-4d9c-a90e-ebc570be3bc6.png)
![Screenshot_20220209-193518](https://user-images.githubusercontent.com/790707/153246618-fe65532d-5125-400e-a9a7-85190d84a5f2.png)
![Screenshot_20220209-193535](https://user-images.githubusercontent.com/790707/153246620-6483b5e2-46b6-4732-8d66-eb2f89628020.png)

**Reply by phhusson** [reply]
Ok. Open a ticket in the "Issues" section
I'll need you to give me the result of `adb logcat -b all -d > log.txt` (attach log.txt to the ticket) after installing the OTA, and the result of `adb shell su 0 find /metadata`

**Reply by RendeRR** [reply]
Done. Thank you. https://github.com/phhusson/treble_experimentations/issues/2185#issue-1130761050

**Reply by dayona** [reply]
@phhusson @RendeRR Is it fixed? Iam having the same issue. It shows update, downloads and install successfully, after reboot system remains same.
If you have solution, please share it.
[phh_ota.txt](https://github.com/phhusson/treble_experimentations/files/8955195/phh_ota.txt)

**Comment by tnolte1** [comment]
Awesome work! The OTA upgrade from v401 to v402 was successful on my Android 10 smartphone.

But when i flashed v401 again to repeat the update, the OTA is not there any more, even when clicking on the "Check for Update" button. Is it not intended to flash an older version or do i just have to wait?

Also, it would be great to be able to see the current System version (like v401 or v402) in Settings->Updater or maybe another place like Settings->System or "About".

**Reply by phhusson** [reply]
Right, so there is indeed the issue that "removing" the ota is complicated. The way this OTA works, is that when you OTAed v402, v401 remained installed, but v401 as soon as it boots, goes into v402 which is stored in data partition. If you flash again v401, since your data partition is kept intact, it'll remain on v402. To clear this, you'll need to remove /metadata/phh/img and/or /data/gsi/phh.

I'll probably add a "delete ota" button somewhere or something, but quite often, downgrading doesn't work (because you're not supposed to access your data with an older SPL than previous one).

**Reply by tnolte1** [reply]
Thanks for you answer, this is more complicated than I thought. Will mixed updates be ok? Like OTA to v402 and then flashing v403?

Anyway as you said downgrading is an unusual case and I did this only to reproduce the following issue: When during the download the screen-saver kicks in, the download aborted and I had to start it again. But not sure if this is really a bug as I cannot reproduct it right now for lack of another OTA.

**Comment by idfake** [comment]
Is there any chance to get the normal gapps version for binder64 back?

**Comment by RFX-LEGEND** [comment]
I'm using a Moto X4 and the sim card isn't being recognized, I had this problem earlier with the pre-release builds too.

**Reply by RFX-LEGEND** [reply]
I opened a ticket for it.

**Comment by Pschynder** [comment]
Boot Loop on Unihertz Jelly 2 previous 400.h booted fine as long as securize was not used

**Comment by martinfutoo** [comment]
Still hoping for aonly vanilla sys32 for Xiaomi Redmi Go.
A 11 works well enough, thanks.

**Reply by NateHubbard** [reply]
You're hoping for Android 12 to support 32 bit?

**Comment by ghost** [comment]
Thanks alot for the awesome work, flashed it on my Galaxy S20 FE SM-G780G and everything works perfectly fine. However, the brightness bar is not working, and the brightness itself is too much dim for some reason

**Reply by tiagogilberto** [reply]
good night friend, I also have a s20 fe, and I would like to know if everything is working perfectly, like the basic things of a cell phone, is it about digital is it working? what version do you download for s20 fe? thanks for the anticipation

**Comment by lovefaruk** [comment]
Not booting with Android Pie on Samsung A70.

**Comment by innofeld** [comment]
Dear Friends
Can we install it on BV-9500-plus smartphone?
As I understood, minimum requirement to install Android 12 is Android 10 + DSU
However can we install it on smartphone with Android 9.0 pie (of course with treble A/B support)? 
Regards

**Reply by devrieshh** [reply]
when the treble check app says / system-as-root supported         it can

**Reply by phhusson** [reply]
>  As I understood, minimum requirement to install Android 12 is Android 10 + DSU

Where did you see that? This is totally wrong

**Comment by mariasanchez84** [comment]
Thanks a lot ! Awesome, everything seems to work perfectly on my Huawei mediapad pro M5, no problem so far. Don't have any 'graphic glitches' what are they for example ?

Logcat is spammed with `W hwcomposer: ScfCanHandle:294: AFBC: afbc layer height scl down larger than 1.8.(src_height:2560, dst_height:1152)` but it doesn't seem to affect battery drain.

**Reply by AbdSattout** [reply]
How you unlocked the Huawei bootloader 
I have **_Huawei Y6 Prime 2018_** but the bootloader locked with a password 
Is there any way for unlocking?

**Reply by mariasanchez84** [reply]
Huawei official website gave me the unlock code when I bought it. Just sell your crap and buy an unlockable device.

**Comment by joaodalvi** [comment]
I have android 11 treble_arm64_bgS-userdebug (don`t know how to find the right version) at my A11 2018 (A750G DS). I want to update it to latest version but can`t find any guide on how to do it. Can some one please point me at the right direction, please?

**Reply by devrieshh** [reply]
Do the treble check app,when it says  / system-as-root supported     you can install   A/B  systems.  the app says also if you have 64bit  or 32bit  with this you can find the right system

**Comment by dpkg-i-foo-deb** [comment]
Motorola One Fusion system-squeak-arm64-ab-vanilla this version doesn't boot anymore, it won't go further than the bootlogo...

**Comment by ghost** [comment]
Can I ask.  I have old Huawei phones China variant with unlock bootloader. But The OS is Harmony os2. I heard the rumors it's actually android Q. Can GSI boot from this 
OS 🧐🤔

**Reply by devrieshh** [reply]
cant you install treble check app to see?

**Reply by ghost** [reply]
Ok
I'll try to check

**Comment by hdk2d** [comment]
not booting motorola one 5g ace

**Reply by hdk2d** [reply]
but Pixel Experience 12.0 v402.1 is booting and bluetooth audio not working 
I will try various Phh treble settings profile settings

**Comment by quoich** [comment]
@phhusson I used the OTA to update from 401 to 402 successfuly. However, the OTA update to 402.1 hasn't worked. It downloads and reboots to 402. I tried flashing 402.1 with fastboot and it didn't work either. Any suggestions?

**Comment by klimentin40** [comment]
Hi, nice work on the GSI's you make, I used AOSP 11 for a while and I really like it, now I want to try AOSP 12 but unfortunately there is no A-Only version of it yet, I was just wondering if there ever will be an A-Only version of AOSP 12?

**Reply by phhusson** [reply]
That's unlikely.

**Reply by klimentin40** [reply]
Oh... I was really hoping for an A-Only version, I guess I'll have to try and find some workarounds for it to work, anyway thanks for replying...

**Comment by martinfutoo** [comment]
Would be very happy to see A-only.

    On Sunday, 27 February 2022, 21:52:52 GMT+7, klimentin40 ***@***.***> wrote:  
 
 


Hi, nice work on the GSI's you make, I used AOSP 11 for a while and I really like it, now I want to try AOSP 12 but unfortunately there is no A-Only version of it yet, I was just wondering if there ever will be an A-Only version of AOSP 12?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by phhusson** [reply]
I too would be very happy to see A-only support.

**Comment by SixTni9** [comment]
![2285DE02-0B1D-4272-82AB-7D78248F8DB2](https://user-images.githubusercontent.com/88974549/156327767-b5d40e3d-5fd9-4f75-bb86-10987a58488e.jpeg) 
Hi developer,

I am having a hard time trying to flash gsi for my configuration. I tried to research but could not found correct information. 
According to my phone info I will need “A-only vndklite” version of gsi. But there is no gsi with this configuration. A small hint can help me a lot! Please 🙏🏻 
I have tried this one but it did not work: https://github.com/phhusson/treble_experimentations/releases/download/v313/system-roar-arm64-aonly-vanilla.img.xz

**Comment by devrieshh** [comment]
your device doesnt support      /system as root   than you cant install A/B  systems  so for now you can only install Aonly systems, no AOSP 12.0  yet

**Reply by SixTni9** [reply]
Is a only vndlite supported ?

**Comment by devrieshh** [comment]
those are the latest Aonly systems

[system-roar-arm64-aonly-floss.img.xz](https://github.com/phhusson/treble_experimentations/releases/download/v313/system-roar-arm64-aonly-floss.img.xz)833 MB
 [system-roar-arm64-aonly-gapps.img.xz](https://github.com/phhusson/treble_experimentations/releases/download/v313/system-roar-arm64-aonly-gapps.img.xz)777 MB
 [system-roar-arm64-aonly-vanilla.img.xz](https://github.com/phhusson/treble_experimentations/releases/download/v313/system-roar-arm64-aonly-vanilla.img.xz)
[system-roar-arm-aonly-gogapps.img.xz](https://github.com/phhusson/treble_experimentations/releases/download/v313/system-roar-arm-aonly-gogapps.img.xz)594 MB
 [system-roar-arm-aonly-vanilla.img.xz](https://github.com/phhusson/treble_experimentations/releases/download/v313/system-roar-arm-aonly-vanilla.img.xz)

**Comment by devrieshh** [comment]
the above systems are   AOSP 11.0 v313

**Reply by devrieshh** [reply]
i guess floss.img in android11  is about the same as  vndlite in android12, no google apps

**Comment by der-lange** [comment]
My device        :  Sony Xperia XZ2P (H8166_Customized DACHBLX_52.1.A.3.49-R6C)
oem-image      : SW_binaries_for_Xperia_Android_12_4.19_v1c_tama.img
system-images: system-squeak-arm64-ab-vndklite-(floss, gapps, vanilla)and                      
                          PixelExperience_arm64-ab-vndklite-12.0-20220208-UNOFFICIAL.img

Installation   : adb reboot bootloader; fastboot -w; fastboot erase system_a;
                       fastboot --disable-verity --disable-verification flash vbmeta vbmeta.img;
                       fastboot flash oem_a xxx.tama.img;
                       fastboot flash system_a system-xxxx.img

Results:         never ending bootloop with PE- and gapps versions
                       floss- and vanillaversion start correctly; but  without oem.img there is no camera-app available
                       and with oem.img "Camera error, Cant connect the camera"

**Reply by phhusson** [reply]
Don't flash OEM partition, this is NOT for use with GSI.
Flash back stock rom. then flash again GSI.

**Comment by devrieshh** [comment]


**Reply by der-lange** [reply]
no change in behavior; all attempts at installation showed: only the FLOSS version boots correctly, but has no camera function

**Comment by Keviny9** [comment]
Add some method of lock screen in settings, causes bootloop in android interface after of a reboot

**Reply by mohammad556655** [reply]
after a reboot, i get infinite bootloop too. is there any way to solve it?

**Comment by GartimusPrime** [comment]
Hello,
I have a Lenovo Tab P11 Pro tablet. When I install any GSI, my auto rotation is backwards. Not much is happening in the XDA page for this device. I was wondering if there is a way to figure out the issue. I would love to start looking at 12 and 12L. I can use them, I just need to use an rotation app from the play store. 
Thank you.

**Reply by yusrang** [reply]
i got same problem here, im using j716f

**Comment by HansDus** [comment]
Hello Pierre

Here is a first report of my experience
In hope it helps you and others...

Divice Info: Samsung Galaxy A505fn
Rom running before Update: AOSP 10.0 v221 from 8 Jul 2020
Rom updated to: AOSP 12.0 v402 from 8 Feb 2022

I did as you told me and it went mostly good.

Using TWRP i did make a Complete Backup of all that you can (Files can not be included)
After flashing the Image onto system only - like you said no wipe needed - it booted up fine and

Viola i am a Happy Android 12 ASOP User now 😁

Looks nice and i like it!
I am going to play around with it the next day and see how it goes

Now a view first observations good and bad some could be fixed some i am still looking how to:

1.) Most Apps work just fine and i Updated many 😁

2.) F-droid worked straight away so i could fix initial problems with Aurora Store by re-Installing it

3.) My Fennec Browser and all its Chronic inkl Desktop links was gone (no big deal for me)
rei-Istalled it and works like a rocket (Newer Version incl)😁

4.) My Banking app started just fine but when i try to lock in it collapses and shuts down
I tryed to de-install it and re-Install it .... Did not worked jet

5.) i noticed the Screen seams to be less responsive and freeze at times
did new start and it seams okay now (i will watch it)

6.) My E-Mail App (Bluemail) unfortunately is also collapsing every-time i try to answer...
in fact i was Trying to write this response on my Phone but did not work hat to resort to my laptop for now I  tryed re-Install as-well did not work

7.) as of now i lost Location service and in the Old version that was all in and worked fine for more that 2 Years without Google - so that,s a Bad one - no solution jet

8.) a Hand full of Apps don't work like usual ore at all and re-Installing seams to help only with some of them

Might be a Idea to Get a Later Version i just Picked this one because it was not Labelled per-Releas, if you think it be better and fix some of my Problems Let me Know...

So that was it for now ...my little Report.
Resume is, it did work but not flawless 😉

I am still Happy about it and will see in the next days how it goes

Best Thank for you,re work and i consider as soon Money comes in to send a Contribution
also for you're friendly and Quick response

Saludos Hans

**Comment by HansDus** [comment]
Hello Pierre

Here is a Update on what i did with the Challenges at hand

I updatet from Version v402 to v411
Flashing v411 to my Device maid most of the Initial Problems vanish away :1st_place_medal: 

1.) All Apps - as far as i tested- are now up and Running just fine :+1:  (Banking, E-Mail and my Favourite Bible-apps etc.)
2.) But Location Services are still not running ( that's really bad some of my Apps because of my Job need it at least sometimes )
3.) Sometimes there seams to be a glitch in the Matrix and my Device would Blank and go back to the Start screen so i have to unlock the phone again ... also it seams to take a moment until the Phone works again.

Resume:
Seams to be the best advice by now for me to Run the latest available Version and off course for those who got a Samsung Galaxy A505x
Its realy Bad that i have no Location-Service at the moment so if anyone here is a crack when it comes to Props with that pleas let me Hear from You!!!

One Question for [ Pierre](https://github.com/phhusson):
As far as i remember in my Version from (2020 AOSP 10.0 v221) there was a Alternative Location-service (not Google) fully operational Preinstalled. If i got that right! What happened to that is that not possible no more?

I prefer to be as Google-free as Possible (lets Kick Big-brother out :-) as much as possible)

If anyone including you know how its done Pleas let me know!!
If it has to be Google location service then at least i want to be anonymous if Possible.
So pleas anyone let me know!
If there is no other way then anyhow would help a lot to know how to get it back

Saludos Hans

Ps: I a, also Considering to Roll back to my initial Backup then go straight to the Latest Version and see what that dose for me!
I keep all in the Loop how that works out!

**Reply by phhusson** [reply]
I'm guessing you're speaking of network-based location, (like indoor) not GPS? GPS shouldn't require anything installed. As for network-based location, I don't remember shipping a Network Location Provider in AOSP 10, but anyway as far as I can tell, you can install one from fdroid. For instance install Apple UnifiedNLP Backend. Then you can enable it in microg settings.

**Reply by HansDus** [reply]
Hello Pierre

Okay last update for today!

I dont know what did it but my Location-service came to live :1st_place_medal: 

All i did i tried to lock in to micro G using a Google-Account (But it was not successful as it said Server Problem Try later)
I thort i can use it for Now and get rid of it later once i have a other solution.
Sometimes Later it just started to work as i hade openMaps open and suddenly it would show me a location so i went and check other apps and now its all there.... Strange ... as i went and check if i am looked in with Google it turns out, no still not locke in Hm

Anyhow still interested if alternative location-service exists

Resume:
As of now I am completely Happy with the result :100:  for now

Saludos Hans

**Reply by HansDus** [reply]
> I'm guessing you're speaking of network-based location, (like indoor) not GPS? GPS shouldn't require anything installed. As for network-based location, I don't remember shipping a Network Location Provider in AOSP 10, but anyway as far as I can tell, you can install one from fdroid. For instance install Apple UnifiedNLP Backend. Then you can enable it in microg settings.

Cool man i will look into it thanks for all the Hints at right time!
Good to know there is alternative Location service.

Like i said Phantastc Job man :1st_place_medal:  I am very happy with you're work and using it for 2 Years now and no plan to change anytime soon, so from now on i follow you and get update :+1:  
As soon my Money is in ill remember you man and send you some 

Saludos Hans

**Comment by HansDus** [comment]
Hello @phhusson 

So here is another view Observations:

As i was Playing around now since Yesterday it turns out all initial problems are solved. :1st_place_medal: 
Having said that there is one last Problem that i fund later.
Its not a very big one but depending on the situation could be really annoying!

At times when you do something in an App suddenly the the screen goes Blank and falls back to the Lock-screen and the device seams unresponsive for a view seconds before you can lock on again.
The app you was working in seams to restart and then work you did is lost 
(example: Text i was just Typing in my Messenger not send jet was gone)

Just to give you notice and askin if you know what it is and how to stop it!

Saludos Hans

**Comment by phhusson** [comment]
Well, that looks like a crash, when it happens, dump logs with `adb logcat -b all -d >log.txt`, open an issue and attach log.txt to the issue

**Reply by HansDus** [reply]
Hm havend Done that before!

Dose it work under Linunx Mint (I only have a win XP running in a virtual Box so i did manage to run Odin 3.14.1 To do some Flashing 2020

After i rootet my own device 2020 i kind of have what i need on my phone and directly flash from SD Card so i lost a little the Knowledge i had 2 years back! I was also Working with a Linux Tool back then but cant remember.

Is there a App that can do the Job to make a Log for you?
Ore can i Use this Funktion with adb under Linux in the Terminal? ( i remeber using adb under Linux)

Also i still need to find out how things work here ( Like open a IIssue ;-)

Saludos Hans

**Reply by HansDus** [reply]
Hallo @phhusson 

Forget my partially senseless Blabbering from the last Post!
Since then i manged using Adb in Linux Terminal to make the Log -file using your exact Command string ;-)

So i indeed have a Log File now i am just a little worried about my Privacy so what exactly do i give away Publicly when i send you this file? 
For example could someone use this information for a Attack on my device?
I don't Know enough to understand fully what this Log Contains!

Saludos Hans

**Reply by HansDus** [reply]
Okay @phhusson 

I did found where to open a Issue and started one here https://github.com/phhusson/treble_experimentations/issues/2240#issue-1179951385

Saludos Hans

**Reply by HansDus** [reply]
> Well, that looks like a crash, when it happens, dump logs with `adb logcat -b all -d >log.txt`, open an issue and attach log.txt to the issue
> […](#)
> On March 24, 2022 3:08:01 PM GMT+01:00, HansDus ***@***.***> wrote: Hello[ Pierre](https://github.com/phhusson) So here is another view Observations: As i was Playing around now since Yesterday it turns out all initial problems are solved. 1st_place_medal Having said that there is one last Problem that i fund later. Its not a very big one but depending on the situation could be really annoying! At times when you do something in an App suddenly the the screen goes Blank and falls back to the Lock-screen and the device seams unresponsive for a view seconds before you can lock on again. The app you was working in seams to restart and then work you did is lost (example: Text i was just Typing in my Messenger not send jet was gone) Just to give you notice and askin if you know what it is and how to stop it! Saludos Hans -- Reply to this email directly or view it on GitHub: [#2174 (comment)](https://github.com/phhusson/treble_experimentations/discussions/2174#discussioncomment-2430431) You are receiving this because you were mentioned. Message ID: ***@***.***>

Hello @phhusson

Now after Two Days Running this Release, i have to say the Problem is sever! (Version v411)
The Phone regularly Crashes as you are in the middle of something makes for example Writing a Text literally like Playing Russian Roulette! SO if you want to write a E-Mail and put a lot of tort in it when it Crashes its Gone because the E-Mail app will restart and what you did is Gone

So i am Unhappy now :-( understandably....
What can be done to stop this?
Is there a Stable Release as far as you Know?

Cant go back to v402 neither to many Bugs concerning Program that don't work anymore.
(remember our earlier exchange on that)
So v411 fixed a lot but now there is only that very annoying Bug that it crashes regularly!!

Pleas also don't forget i need reassurance what this Log file contains and if it can be used to harm my Privacy ore security my Device in any way shape or form when i post it Publicly here!?

Pleas Help i realy Like Your Google-free Custum-Rom

Saludos Hans

**Comment by Rafitu** [comment]
> * More February security patch level (previous one had only some patches, not all)
> 
>     * Well this will allow users to test OTA on their devices
> 
> 
> _This discussion was created from the release [AOSP 12.0 v402](https://github.com/phhusson/treble_experimentations/releases/tag/v402).
In Cubot kingkong mini 2 it seems to work fine. Thank you so much.

**Comment by lbrunoz** [comment]
All phh roms with android 12 work almost perfectly but on my moto z2 force (nash) phone I have two problems a bug in offline loading where the battery percentage even appears but after a while it starts to crash and out of nowhere it starts to run the boot animation (it loads normally, but you can't see the load progress)
and second that in all roms with android 12 the sim card does not recognize and I have no idea how to fix it

**Comment by dayona** [comment]
Hi @phhusson .. its urgent . please reply. I add all the necessary changes for ota updater on android 12 and it works perfectly fine. But when I add the same changes on android 11 , it does not work.

"phh mount returned -1 " is the log on rebooting after update on android 11.

I think
'int mountRes = mount("/dev/root-otaphh", "/debug_ramdisk", "ext4", MS_RDONLY, ""); '
doesn't work properly on android11. If you or anyone have a way to fix it, please reply

**Reply by phhusson** [reply]
Hi, my consultancy fee for "urgent" is 400€/hour
